runtime/oti/j9nonbuilder.h

```diff
@@ -2294,11 +2294,14 @@ typedef struct J9ROMMethodHandleRef {
 #define MN_TRUSTED_FINAL	0x00200000
 #define MN_HIDDEN_MEMBER	0x00400000
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-#define MN_FLATTENED		0x00800000
-#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
-
+#define MN_FLAT_FIELD		0x00800000
+#define MN_NULL_RESTRICTED	0x01000000
+#define MN_REFERENCE_KIND_SHIFT		26
+#define MN_REFERENCE_KIND_MASK		0x3C000000 >> MN_REFERENCE_KIND_SHIFT	/* (flag >> MN_REFERENCE_KIND_SHIFT) & MN_REFERENCE_KIND_MASK */
```

Comment:
Please add parentheses here:
```c
#define MN_REFERENCE_KIND_MASK		(0x3C000000 >> MN_REFERENCE_KIND_SHIFT))
```
On the other hand, the result is the same (`0xF`) when `!defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)`: why make the definition conditonal?

Reply:
> On the other hand, the result is the same (0xF) when !defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES): why make the definition conditional?

As per https://github.com/eclipse-openj9/openj9/blob/2c34fcf0fb16edf7ff0c5af17db501e6107cb39f/runtime/oti/j9nonbuilder.h#L2285-L2286
This maps https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes/blob/12c6a4c77383821a4499e5472d7c4241542d546f/src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java#L126-L129
```
            MN_FLAT_FIELD            = 0x00800000, // flat field
            MN_NULL_RESTRICTED       = 0x01000000, // null-restricted field
            MN_REFERENCE_KIND_SHIFT  = 26, // refKind
            MN_REFERENCE_KIND_MASK   = 0x3C000000 >> MN_REFERENCE_KIND_SHIFT;
```

Added parentheses at 
* https://github.com/eclipse-openj9/openj9/pull/19446